aptos-move/framework/aptos-stdlib/doc/smart_vector.md

```diff
@@ -0,0 +1,750 @@
+
+<a name="0x1_smart_vector"></a>
+
+# Module `0x1::smart_vector`
+
+
+
+-  [Struct `SmartVector`](#0x1_smart_vector_SmartVector)
+-  [Constants](#@Constants_0)
+-  [Function `empty`](#0x1_smart_vector_empty)
+-  [Function `empty_with_config`](#0x1_smart_vector_empty_with_config)
+-  [Function `singleton`](#0x1_smart_vector_singleton)
+-  [Function `destroy_empty`](#0x1_smart_vector_destroy_empty)
+-  [Function `borrow`](#0x1_smart_vector_borrow)
+-  [Function `borrow_mut`](#0x1_smart_vector_borrow_mut)
+-  [Function `append`](#0x1_smart_vector_append)
+-  [Function `push_back`](#0x1_smart_vector_push_back)
+-  [Function `pop_back`](#0x1_smart_vector_pop_back)
+-  [Function `remove`](#0x1_smart_vector_remove)
+-  [Function `swap_remove`](#0x1_smart_vector_swap_remove)
+-  [Function `swap`](#0x1_smart_vector_swap)
+-  [Function `reverse`](#0x1_smart_vector_reverse)
+-  [Function `index_of`](#0x1_smart_vector_index_of)
+-  [Function `contains`](#0x1_smart_vector_contains)
+-  [Function `length`](#0x1_smart_vector_length)
+-  [Function `is_empty`](#0x1_smart_vector_is_empty)
+
+
+<pre><code><b>use</b> <a href="big_vector.md#0x1_big_vector">0x1::big_vector</a>;
+<b>use</b> <a href="../../move-stdlib/doc/error.md#0x1_error">0x1::error</a>;
+<b>use</b> <a href="math64.md#0x1_math64">0x1::math64</a>;
+<b>use</b> <a href="../../move-stdlib/doc/option.md#0x1_option">0x1::option</a>;
+<b>use</b> <a href="type_info.md#0x1_type_info">0x1::type_info</a>;
+<b>use</b> <a href="../../move-stdlib/doc/vector.md#0x1_vector">0x1::vector</a>;
+</code></pre>
+
+
+
+<a name="0x1_smart_vector_SmartVector"></a>
+
+## Struct `SmartVector`
+
+A Scalable vector implementation based on tables, elements are grouped into buckets with <code>bucket_size</code>.
+The vector wrapping BigVector represents an option w/o <code>Drop</code> ability requirement of T to save space of the
```

Comment:
Suggest:

The vector wrapping BigVector represents an option w/o the <code>Drop</code> ability requirement of T to save space in the metadata associated with BigVector when smart_vector is so small that inline_vec vector can hold all the data.